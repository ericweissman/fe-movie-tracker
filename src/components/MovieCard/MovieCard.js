import React from "react"
import PropTypes from "prop-types"
import { Link } from "react-router-dom"

const MovieCard = ({ movie, user, handleDelete, handleFavorite }) => {
  const { movie_id, poster_path } = movie
  const movieImageUrl = "https://image.tmdb.org/t/p/w500"
  return (
    <div className="movie-card-div">
      <Link to={`/movies/${movie_id}`}>
        <img src={movieImageUrl + poster_path} alt="movie poster" />
      </Link>
      {user.id && user.favorites.includes(movie_id) && (
        <button onClick={() => handleDelete(movie, user)}>DELETE FROM FAVORITES</button>
      )}
      {user.id && !user.favorites.includes(movie_id) && (
        <button onClick={() => handleFavorite(movie, user)}>ADD TO FAVORITES</button>
      )}
      {!user.id && (
        <Link to="/login">
          <button> ADD TO FAVORITES</button>
        </Link>
      )}
    </div>
  )
}

export default MovieCard


MovieCard.propTypes = {
  handleDelete: PropTypes.func.isRequired,
  handleFavorite: PropTypes.func.isRequired,
  movies: PropTypes.array,
  user: PropTypes.object.isRequired
}
