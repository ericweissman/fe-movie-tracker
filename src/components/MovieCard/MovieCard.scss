.movie-card-div {
    height: 500px;
    width: 325px;
    border-radius: 15px;

  img {
    height: 500px;
    width: 325px;
    border-radius: 15px;
  }
  button {
    position: relative;
    z-index: 1;
    bottom: 50px;
  }
}
