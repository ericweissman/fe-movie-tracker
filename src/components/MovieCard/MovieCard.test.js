import React from "react"
import MovieCard from "../MovieCard/MovieCard"
import { shallow } from "enzyme"

let wrapper;
const handleDeleteMock = jest.fn();
const handleFavoriteMock = jest.fn();
const movieMock = {
  vote_count: 3078,
  movie_id: 297802,
  video: false,
  vote_average: 6.9,
  title: "Aquaman",
  popularity: 805.109,
  poster_path: "/5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg",
  original_language: "en",
  original_title: "Aquaman"
}
const userMock = { name: 'Hill', id: 2, favorites: [297802]}

describe('MovieCard', () => {

  it('should match snapshot', () => {
    wrapper = shallow(<MovieCard 
      key={movieMock.id}
      movie={movieMock}
      handleFavorite={handleFavoriteMock}
      user={userMock}
      handleDelete={handleDeleteMock} />)
      expect(wrapper).toMatchSnapshot()
  });

  it('should call handleDelete when the delete button is clicked', () => {
    wrapper.find('.delete').simulate('click');
    expect(handleDeleteMock).toHaveBeenCalled();
  });

  it('should match snapshot if user has no favorites or if there is no user', () => {
    const user2 = {};
    wrapper = shallow(<MovieCard 
      key={movieMock.id}
      movie={movieMock}
      handleFavorite={handleFavoriteMock}
      user={user2}
      handleDelete={handleDeleteMock} />);
      expect(wrapper).toMatchSnapshot();
  })

  it('should call handleFavorite when the favorite button is clicked and there is a user without the favorite in the array', () => {
    const user3 = {name: 'Christy', id: 5, favorites: []}
    wrapper = shallow(<MovieCard 
        key={movieMock.id}
        movie={movieMock}
        handleFavorite={handleFavoriteMock}
        user={user3}
        handleDelete={handleDeleteMock} />);
    wrapper.find('.favorite').simulate('click');
    expect(handleFavoriteMock).toHaveBeenCalled();
    });

  it('should match the correct snapshot when there is a user without favorites', () => {
    expect(wrapper).toMatchSnapshot();
  });
});