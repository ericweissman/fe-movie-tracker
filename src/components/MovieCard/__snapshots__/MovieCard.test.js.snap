// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match snapshot 1`] = `
<div
  className="movie-card-div"
>
  <Link
    replace={false}
    to="/movies/297802"
  >
    <img
      alt="movie poster"
      src="https://image.tmdb.org/t/p/w500/5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg"
    />
  </Link>
  <button
    className="delete"
    onClick={[Function]}
  >
    DELETE
  </button>
</div>
`;

exports[`MovieCard should match snapshot if user has no favorites or if there is no user 1`] = `
<div
  className="movie-card-div"
>
  <Link
    replace={false}
    to="/movies/297802"
  >
    <img
      alt="movie poster"
      src="https://image.tmdb.org/t/p/w500/5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg"
    />
  </Link>
  <Link
    replace={false}
    to="/login"
  >
    <button>
      FAVORITE
    </button>
  </Link>
</div>
`;

exports[`MovieCard should match the correct snapshot when there is a user without favorites 1`] = `
<div
  className="movie-card-div"
>
  <Link
    replace={false}
    to="/movies/297802"
  >
    <img
      alt="movie poster"
      src="https://image.tmdb.org/t/p/w500/5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg"
    />
  </Link>
  <button
    className="favorite"
    onClick={[Function]}
  >
    FAVORITE
  </button>
</div>
`;
